src/proto/uprotocol/uauthority.rs

```diff
@@ -11,16 +11,104 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+use bytes::BufMut;
+
 use crate::uprotocol::uri::uauthority::Number;
 use crate::uprotocol::UAuthority;
 
-pub use crate::uri::validator::ValidationError;
+pub use crate::proto::uprotocol::uuri::SerializationError;
+use crate::uri::validator::ValidationError;
 
 const REMOTE_IPV4_BYTES: usize = 4;
 const REMOTE_IPV6_BYTES: usize = 16;
 const REMOTE_ID_MINIMUM_BYTES: usize = 1;
 const REMOTE_ID_MAXIMUM_BYTES: usize = 255;
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum AddressType {
```

Comment:
You might want to consider adding
```rust
#[repr(u8)]
pub enum AddressType
```
FMPOV you would then not need to cast the values to `u8` in the `value` function ...

Reply:
The #[repr(u8)] is a good idea - I just don't think we can get rid of the 'as u8', as the value() method is casting from `AddressType` to u8, regardless of what size AddressType is.